src/ripple/protocol/impl/SField.cpp

```diff
@@ -296,12 +296,12 @@ SField::SField (SerializedTypeID tid, int fv)
         : fieldCode (field_code (tid, fv))
         , fieldType (tid)
         , fieldValue (fv)
+        , fieldName (std::to_string (tid) + '/' + std::to_string (fv))
         , fieldMeta (sMD_Default)
         , fieldNum (++num)
         , signingField (IsSigning::yes)
+        , jsonName (fieldName.c_str())
```

Comment:
If this constructor, or the other constructor which points `jsonName` into `fieldName` is used, and then that `SField` is move constructed, then because of the defaulted `SField` move constructor, `jsonName` will point into a destructed object.

I think this patch is getting away with it because all of the move constructions of `SField` are RVO'd away.  However that's a guess, and not something we should depend on anyway.  I think we should implement the `SField` move constructor to detect the self-referencing `jsonName` pointer and update it when necessary.